types/webpack/index.d.ts

```diff
@@ -646,6 +648,118 @@ declare namespace webpack {
+             * devServer.publicPath will be used to determine where the bundles should be served from, and takes precedence.
+             */
+            contentBase?: boolean | string | string[];
+            /**
+             * When set to true this option bypasses host checking.
+             * THIS IS NOT RECOMMENDED as apps that do not check the host are vulnerable to DNS rebinding attacks.
+             */
+            disableHostCheck?: boolean;
+            /**
+             * This option lets you reduce the compilations in lazy mode.
+             * By default in lazy mode, every request results in a new compilation.
+             * With filename, it's possible to only compile when a certain file is requested.
+             */
+            filename?: string;
+            /** Adds headers to all responses. */
+            headers?: object;
+            /** When using the HTML5 History API, the index.html page will likely have to be served in place of any 404 responses. */
+            historyApiFallback?: boolean | object;
+            /** Specify a host to use. By default this is localhost. */
+            host?: string;
+            /** Enable webpack's Hot Module Replacement feature. */
+            hot?: boolean;
+            /** Enables Hot Module Replacement (see devServer.hot) without page refresh as fallback in case of build failures. */
+            hotOnly?: boolean;
+            /** By default dev-server will be served over HTTP. It can optionally be served over HTTP/2 with HTTPS. */
+            https?: boolean | object;
+            /** The filename that is considered the index file. */
+            index?: string;
+            /**
+             * Toggle between the dev-server's two different modes.
+             * By default the application will be served with inline mode enabled.
+             * This means that a script will be inserted in your bundle to take care of live reloading,
+             * and build messages will appear in the browser console.
+             */
+            inline?: boolean;
+            /**
+             * When lazy is enabled, the dev-server will only compile the bundle when it gets requested.
+             * This means that webpack will not watch any file changes.
+             */
+            lazy?: boolean;
+            /**
+             * With noInfo enabled, messages like the webpack bundle information that is shown
+             * when starting up and after each save,will be hidden.
+             * Errors and warnings will still be shown.
+             */
+            noInfo?: boolean;
+            /** When open is enabled, the dev server will open the browser. */
+            open?: boolean;
+            /** Specify a page to navigate to when opening the browser. */
+            openPage?: string;
+            /** Shows a full-screen overlay in the browser when there are compiler errors or warnings. Disabled by default. */
+            overlay?: boolean | object;
```

Comment:
There than object create an interface with the valid options 

https://webpack.js.org/configuration/dev-server/#devserver-overlay